hue-light-simulator.ino

```diff
@@ -1,67 +1,73 @@
-boolean gPowerState = true;
+#include "math.h"
+
+int gPowerState = 1;
 float gHue = 1;
 float gSaturation = 1;
 float gBrightness = 1;
 
-void setup()
-{
-   RGB.control(true);
+int redPin = A0;    // RED pin of the LED to PWM pin **A0**
+int greenPin = D0;  // GREEN pin of the LED to PWM pin **D0**
+int bluePin = D1;   // BLUE pin of the LED to PWM pin **D1**
 
-   updateLED();
+void setup() {
+   pinMode(redPin, OUTPUT);
+   pinMode(greenPin, OUTPUT);
+   pinMode(bluePin, OUTPUT);
 
    //Exposed internal functions as cloud API
-   Spark.function("powerState", powerState);
-   Spark.function("hue", hue);
-   Spark.function("saturation", saturation);
-   Spark.function("brightness", brightness);
+   Particle.function("powerState", powerState);
+   Particle.function("hue", hue);
+   Particle.function("saturation", saturation);
+   Particle.function("brightness", brightness);
+}
+
+void loop() {
+    /* Do nothing */
 }
 
 /* Cloud methods */
 
 int powerState(String value) {
-    int result = -1;
-
+    int result = 1;
     if (value.length() > 0) {
         gPowerState = (value.toInt() == 1);
         result = gPowerState;
     }
-
     updateLED();
-
     return result;
 }
 
 int hue(String value) {
+    delay(200);
```

Comment:
Needs testing without the delays. 
If I use the slider to change colour in Home then the Core crashes.